src/languages/smtlib2/poly/parser.mly

```diff
@@ -127,6 +128,12 @@ sort:
         T.const ~loc (f I.sort)
       in
       let loc = L.mk_pos $startpos $endpos in T.apply ~loc c args }
+  | OPEN ARROW h=sort t=sort+ CLOSE
+    { let loc = L.mk_pos $startpos $endpos in
+      let rev_t = List.rev t in
+      let ret = List.hd rev_t in
+      let args = h :: List.rev rev_t in
```

Comment:
Two remarks:
- for correctness, you're missing a `List.tl rev_t`
- for code cleanliness (and efficiency), you should actually define a function `split_last` in `src/standard/misc.ml` and call it from here, and in that function, avoid the double `rev`

Reply:
Done